code/game/objects/item_materials.dm

```diff
@@ -1,10 +1,12 @@
 /obj/item/on_update_icon()
-	overlays.Cut()
+	. = ..()
+	SHOULD_CALL_PARENT(TRUE)
```

Comment:
That's because there's a problem here with the idea of slapping in a SHOULD_CALL_PARENT like you suggested last time in discord. I mentioned it in the previous comment on the PR. 
/atom/on_update_icon() is a stub. We don't want to call the stub since its a waste of time. But the SHOULD_CALL_PARENT macro makes it so /obj/item/on_update_icon() has to call the parent stub.. The objective was just to force children of /obj/item to call their parents to avoid problems.